apps/studio/src/pages/home/library/MarketplaceTab/CreateSale.tsx

```diff
@@ -134,16 +147,20 @@ export const CreateSale = () => {
           submitForm,
           setTouched,
         }) => {
-          const perTokenPrice = totalSaleValue / tokensToSell;
+          // tokensToSell is "" when the field is empty, converted to 0
+          const perTokenPrice =
+            Number(tokensToSell) !== 0 ? totalSaleValue / tokensToSell : 0;
           const percentageOfUserStreamTokens =
             (tokensToSell / streamTokensInWallet) * 100;
           const formattedPercentageOfUserStreamTokens = `${percentageOfUserStreamTokens.toFixed(
-            2
+            Math.max(2, 7 - tokensToSell.toString().length)
```

Comment:
This seemed like the most logical dynamic handling of the percentage as it covers up to the 1/(max user tokens).